### PR TITLE
Update

### PR DESCRIPTION
**Pull request by DominiqueMakowski**

### CODE REVIEW

**Comment by strengejacke** [comment]
Should we add an option in `standardize()` to also allow standardizing factors (by converting them to numeric)? I added two helper functions that might be useful in this context... I often got requests to have such a feature in `sjmisc::std()`, so we should also have it here. I'm not sure about the name of the argument that enables this option, though...

**Comment by mattansb** [comment]
`factor_as_numeric`? To indicate the factors' coefs should be treated as numeric?

**Comment by strengejacke** [comment]
``` r
library(parameters)

standardize(iris$Species)
#>   [1] setosa     setosa     setosa     setosa     setosa     setosa    
#>   [7] setosa     setosa     setosa     setosa     setosa     setosa    
#>  [13] setosa     setosa     setosa     setosa     setosa     setosa    
#>  [19] setosa     setosa     setosa     setosa     setosa     setosa    
#>  [25] setosa     setosa     setosa     setosa     setosa     setosa    
#>  [31] setosa     setosa     setosa     setosa     setosa     setosa    
#>  [37] setosa     setosa     setosa     setosa     setosa     setosa    
#>  [43] setosa     setosa     setosa     setosa     setosa     setosa    
#>  [49] setosa     setosa     versicolor versicolor versicolor versicolor
#>  [55] versicolor versicolor versicolor versicolor versicolor versicolor
#>  [61] versicolor versicolor versicolor versicolor versicolor versicolor
#>  [67] versicolor versicolor versicolor versicolor versicolor versicolor
#>  [73] versicolor versicolor versicolor versicolor versicolor versicolor
#>  [79] versicolor versicolor versicolor versicolor versicolor versicolor
#>  [85] versicolor versicolor versicolor versicolor versicolor versicolor
#>  [91] versicolor versicolor versicolor versicolor versicolor versicolor
#>  [97] versicolor versicolor versicolor versicolor virginica  virginica 
#> [103] virginica  virginica  virginica  virginica  virginica  virginica 
#> [109] virginica  virginica  virginica  virginica  virginica  virginica 
#> [115] virginica  virginica  virginica  virginica  virginica  virginica 
#> [121] virginica  virginica  virginica  virginica  virginica  virginica 
#> [127] virginica  virginica  virginica  virginica  virginica  virginica 
#> [133] virginica  virginica  virginica  virginica  virginica  virginica 
#> [139] virginica  virginica  virginica  virginica  virginica  virginica 
#> [145] virginica  virginica  virginica  virginica  virginica  virginica 
#> Levels: setosa versicolor virginica

standardize(iris$Species,force = TRUE)
#>   [1] -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656
#>   [8] -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656
#>  [15] -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656
#>  [22] -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656
#>  [29] -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656
#>  [36] -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656
#>  [43] -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656 -1.220656
#>  [50] -1.220656  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000
#>  [57]  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000
#>  [64]  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000
#>  [71]  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000
#>  [78]  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000
#>  [85]  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000
#>  [92]  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000
#>  [99]  0.000000  0.000000  1.220656  1.220656  1.220656  1.220656  1.220656
#> [106]  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656
#> [113]  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656
#> [120]  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656
#> [127]  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656
#> [134]  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656
#> [141]  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656  1.220656
#> [148]  1.220656  1.220656  1.220656
```

<sup>Created on 2019-05-03 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by DominiqueMakowski** [comment]
Out of curiosity, I wonder in which case you'd want to standardize factors (assuming they are "true" non-ordinal factors)? Do you know the context of such request in sjmisc?

**Comment by DominiqueMakowski** [comment]
```
W  checking S3 generic/method consistency (817ms)
   standardize:
     function(x, robust, method, verbose, force, ...)
   standardize.character:
     function(x, force, ...)
   
   standardize:
     function(x, robust, method, verbose, force, ...)
   standardize.data.frame:
     function(x, robust, method, select, exclude, verbose, force, ...)
   
   standardize:
     function(x, robust, method, verbose, force, ...)
   standardize.factor:
     function(x, force, ...)
   
   standardize:
     function(x, robust, method, verbose, force, ...)
   standardize.grouped_df:
     function(x, robust, method, select, exclude, verbose, force, ...)
   
   See section 'Generic functions and methods' in the 'Writing R
   Extensions' manual.
```

What is the easiest way to resolve this?

**Comment by strengejacke** [comment]
The easiest is to define the generic with only the first argument and `...`, and then use `@rdname` for all class-methods like `standardize.numeric()` etc.

File: R/model_parameters.bayesian.R [hunk 1]
```diff
@@ -16,8 +16,12 @@
   parameters <- .extract_parameters_bayesian(model, ci, ci_method = ci_method, estimate = tolower(estimate), test = test, rope_range = rope_range, iterations = iterations, ...)
 
   # Standardized
-  if (standardize) {
-    std_parameters <- standardize_parameters(model, estimate = tolower(estimate), ...)
+  if (standardize != FALSE & !is.null(standardize)) {
```
[the remark this answers is not in the document]
Comment (DominiqueMakowski) [reply R/model_parameters.bayesian.R]: How to address?

File: R/model_parameters.merMod.R [hunk 2]
```diff
@@ -21,12 +21,13 @@ model_parameters.merMod <- function(model, ci = .95, standardize = FALSE, bootst
...
-    names(std_parameters) <- paste0("Std_", names(std_parameters))
-
-    parameters <- cbind(parameters, std_parameters[c("Std_beta", "Std_SE")])
+  # Standardized
+  if (standardize != FALSE & !is.null(standardize)) {
+    if(standardize == TRUE){
+      warning("Please set the `standardize` method explicitly. Set to \"refit\" by default.")
+      standardize <- "refit"
+    }
+    parameters <- cbind(parameters, standardize_parameters(model, method = standardize)[2])
```
Comment (strengejacke) [inline R/model_parameters.merMod.R]: I mainly revised these code-chunks into:

```r
  # Standardized
  if (isTRUE(standardize)) {
    warning("Please set the `standardize` method explicitly. Set to \"refit\" by default.")
    standardize <- "refit"
  }

  if (!is.null(standardize) && !is.logical(standardize)) {
    parameters <- cbind(parameters, standardize_parameters(model, method = standardize)[2])
  }
```

`isTRUE()` captures also the "logical > 1 coercion"-problem in if-statements. And if `standardize = FALSE`, the 2nd if-condition is not true, so it should work as expected.

**Comment by DominiqueMakowski** [comment]
I reckon the `model_parameters` code could be somehow reorganized, although I don't know how, as several chunks of code a redundant (e.g., the standardize chunk in model_parmaeters that repeats for lms, glms and glmms) 😕

File: R/model_parameters.merMod.R (hunk 2 above)
Comment (DominiqueMakowski) [reply R/model_parameters.merMod.R]: great

**Comment by DominiqueMakowski** [comment]
I think I'll need to update `report` to work with these changes, then we can merge this :)

**Comment by strengejacke** [comment]
The next time I have a bit less time to work on packages, but I think as long as it works we can do the code optimization later.

**Comment by DominiqueMakowski** [comment]
yup it's more a second phase thing :)

**Pull request merged**